### PR TITLE
[Cheat Sheets] Add Icon in the Help Menu

### LINKED ISSUE #1291: [Cheat Sheets] Icon is missing in the Help Menu
Let's make sure issue is not already fixed in latest builds first.

- [x] I verified [latest Integration Build of Eclipse SDK](https://download.eclipse.org/eclipse/downloads/) doesn't provide the feature.

## Suggestion

Currently, the "Cheat Sheets..." Help Menu entry does not have an Icon:
grafik

I think It would be good to add the existing Cheat Sheet Icon there, as it makes the menu entry more visible:
grafik
Also, the Icon is already there, It's just not used in the menu.

## Community

- [x] I understand suggesting an enhancement doesn't mandate anyone to implement it. Other contributors may consider this suggestion, or not, at their own convenience. The most efficient way to get it fixed is that I implement it myself and contribute it back as a good quality patch to the project.

### PR DESCRIPTION
**Pull request by N1k145**
fixes #1291

Adds the existing Cheat Sheet Icon to the "Cheat Sheets.." Help Menu Entry

### CODE REVIEW

**Comment by merks** [comment]
@BeckerWdf @N1k145 

It's not so common to reference images from another bundle:

image

Even some of the above aren't actually referencing another bundle's image.

I wonder too about the use of a "view" image on the menu.  Might that be problematic in the future?

I realize it's quite heavy weight to introduce an icon to this bundle, and to the images repository...

**Comment by N1k145** [comment]
@merks Thank you for your Insight.

For me, it was not uncommon to do it like this, especially as there is already a dependency between these bundles. 

My Intention was to not duplicate the image, as they are already present in the Cheat Sheet bundle.

I was also wondering why this is implemented in the platform bundle and not the Cheat Sheet bundle, but the discussion here gave some insight https://bugs.eclipse.org/bugs/show_bug.cgi?id=120087

If the "view" image will be an issue, IDK. I chose that image because it's the main image which is associated with Cheat Sheets, as it is the view icon.

**Comment by N1k145** [comment]
@BeckerWdf I duplicated the icon and put it to the other images in the plugin

**Comment by merks** [comment]
@BeckerWdf 

From experience I know that such images originate from here:

https://github.com/eclipse-platform/eclipse.platform.images/blob/master/org.eclipse.images/eclipse-svg/org.eclipse.ui.cheatsheets/icons/view16/cheatsheet_view.svg

I also know there is a pretty "strict" folder naming convention involved so that one knows the role the image plays.

https://github.com/eclipse-platform/eclipse.platform.images/tree/master/org.eclipse.images/eclipse-svg/org.eclipse.ui.cheatsheets/icons

Is that convention documented somewhere?  Also, will images need to be copied in the https://github.com/eclipse-platform/eclipse.platform.images repository to the appropriately named folder location?

@N1k145 

Sorry, I'm not trying to create more work for you.  😞   I'm just trying to avoid creating work and surprises for someone else in the future if there are efforts to provide newer images in the future.

**Comment by merks** [comment]
I'm going to leave it to @BeckerWdf  to review and approve.  I'm just trying to be helpful, but traveling for most of the week and would only hold things up.

**Review by BeckerWdf — CHANGES_REQUESTED** [review]

File: platform/org.eclipse.platform/images/cheatsheet_view.png
Comment (N1k145) [reply platform/org.eclipse.platform/images/cheatsheet_view.png:1]: Done, should I leave them in the images folder or move them to a new icons/obj16 folder?

Comment (merks) [reply platform/org.eclipse.platform/images/cheatsheet_view.png:1]: There aren't many images folders:
```
$ls */images
org.eclipse.e4.ui.workbench.swt/images:
dragHandle-rotated.svg  dragHandle.svg

org.eclipse.jdt/images:
topiclabel/

org.eclipse.pde/images:
topiclabel/

org.eclipse.platform/images:
topiclabel/
```

I would find one of the other icons shown on the menus, and use the same folder name convention.  E.g., for the Help Contents it's like this:

https://github.com/eclipse-platform/eclipse.platform.ui/blob/41e1e3ffcccd9c4253f76d68bf5324a8092eac6b/bundles/org.eclipse.ui/plugin.xml#L1516-L1519

I suppose the $nls$ is for translations to provide different icons...

Comment (BeckerWdf) [reply platform/org.eclipse.platform/images/cheatsheet_view.png:1]: mh. pls create a new folder according to https://eclipse-platform.github.io/ui-best-practices/#folder_structure 
(i think "/icons/tool" would be the correct location) and also keep the same structure in the images repo.

Comment (BeckerWdf) [reply platform/org.eclipse.platform/images/cheatsheet_view.png:1]: > I would find one of the other icons shown on the menus, and use the same folder name convention. E.g., for the Help Contents it's like this:
> 
> https://github.com/eclipse-platform/eclipse.platform.ui/blob/41e1e3ffcccd9c4253f76d68bf5324a8092eac6b/bundles/org.eclipse.ui/plugin.xml#L1516-L1519

that's also ok for me.

>I suppose the 
 is for translations to provide different icons...
 
 yes

Comment (N1k145) [reply platform/org.eclipse.platform/images/cheatsheet_view.png:1]: @merks @BeckerWdf 
I moved the icons now to /icons/etool16/ and added the translation tag

**Review by BeckerWdf — APPROVED** [review]

**Comment by BeckerWdf** [comment]
LGTM. I'll merge once CI is happy with this change.

**Comment by BeckerWdf** [comment]
Test org.eclipse.ua.tests.doc.internal.linkchecker.LinkTest is failing. Can this be related to this change? I don't think so.

**Comment by BeckerWdf** [comment]
is it necessary to add the new "icons" folder to the build.properties file in the "bin.includes" key?

**Comment by iloveeclipse** [comment]
> is it necessary to add the new "icons" folder to the build.properties file in the "bin.includes" key?

Sure. What's nit included will be not in the bundle.

**Comment by N1k145** [comment]
@BeckerWdf @iloveeclipse thanks, no idea how I missed that...

**Comment by merks** [comment]
At development time all the content visible in the root folder are visible in the self-hosted launch bundle so it is easy to overlook.

**Comment by BeckerWdf** [comment]
> At development time all the content visible in the root folder are visible in the self-hosted launch bundle so it is easy to overlook.

I *always* oversee this. ;-)

**Review by BeckerWdf — APPROVED** [review]

**Comment by BeckerWdf** [comment]
failing tests

```
[org.eclipse.core.tests.internal.builders.ParallelBuildChainTest.testIndividualProjectBuilds_WithManyProjects_ProjectRelaxedRule](https://ci.eclipse.org/platform/job/eclipse.platform/job/PR-1292/6/testReport/junit/org.eclipse.core.tests.internal.builders/ParallelBuildChainTest/testIndividualProjectBuilds_WithManyProjects_ProjectRelaxedRule/)
[org.eclipse.ua.tests.doc.internal.linkchecker.LinkTest.testAllLinks](https://ci.eclipse.org/platform/job/eclipse.platform/job/PR-1292/6/testReport/junit/org.eclipse.ua.tests.doc.internal.linkchecker/LinkTest/testAllLinks/)

```
are unrelated

**Comment by BeckerWdf** [comment]
@N1k145 thanks for your contribution

**Pull request merged**
